lib/rubygems/specification_policy.rb

```diff
@@ -173,13 +173,18 @@ def validate_duplicate_dependencies # :nodoc:
   end
 
   ##
+  # Checks that dependencies are not self referencing itself.
   # Checks that dependencies use requirements as we recommend.  Warnings are
   # issued when dependencies are open-ended or overly strict for semantic
   # versioning.
 
   def validate_dependencies # :nodoc:
     warning_messages = []
     @specification.dependencies.each do |dep|
+      if dep.name == @specification.name # warn on self reference
+        warning_messages << "Self referencing dependency is highly not recommended and can cause resolving issues."
```

Comment:
I'd go with

```suggestion
        warning_messages << "Self referencing dependency is unnecessary and not recommended."
```

Reply:
Or even "strongly discouraged" instead of "not recommended".

Reply:
> Or even "strongly discouraged" instead of "not recommended".

Yes, like this one https://rubygems.org/gems/standalone_migrations/versions/2.0.4.